### ISSUE TITLE
is your code is about  Why Random Reshuffling Beats Stochastic Gradient Descent

### ISSUE THREAD

**Issue by Sandy4321** [konstmish/random_reshuffling#1]
great code thanks

is your code is about 
https://arxiv.org/pdf/1510.08560.pdf
Why Random Reshuffling Beats
Stochastic Gradient Descent

do you know other python implementations for RR
for example for sparse big data tabular data regression or
sparse big data tabular data logistic regression (classification ) ?

**Comment by konstmish** [comment]
Sorry for the delay in the reply. The code here was written for a different paper (https://arxiv.org/abs/2006.05988), but it is highly related to the one that you mentioned, in particular, the method (Random Reshuffling) is the same in both papers. 

If you're interested in an efficient implementation of RR, you can check https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html
I think you need to provide argument `loss="log"` to use logistic regression loss. If I'm not mistaken, it's written in a mixture of Python and Cython.

Another option is to use the default data loaders in PyTorch, which typically permute the date. You can find many resources online on how to implement logistic regression in PyTorch. Unfortunately, I can't point to a specific reference.
